component/src/main/java/org/ballerinalang/ftp/client/endpoint/InitEndpoint.java

```diff
@@ -68,8 +68,12 @@ public void execute(Context context) {
                 password = basicAuth.getStringField(FtpConstants.ENDPOINT_CONFIG_PASSWORD);
             }
         }
-        String url = FTPUtil.createUrl(protocol.getStringValue(), host, port, username, password, null);
```

Comment:
Why we remove this? Previously we constructed the URL only once. But now, it will create every time when the action invoked.

Reply:
Earlier, URI were built using String builder. For this, We did the null check for every variables and build the URI without nullable variable. If we create the URI as above, we'll get the error when username or password contains certain values(@ , - and etc).  
So I created the URI from uri = new URI(). It didn't support nullable variable. Here, Base path is null. So I have removed.